prefs.js

```diff
@@ -259,8 +247,9 @@ var Settings = class DashToDock_Settings {
 
         this._monitors = [];
         // Build options based on the number of monitors and the current settings.
-        let n_monitors = Gdk.Screen.get_default().get_n_monitors();
-        let primary_monitor = Gdk.Screen.get_default().get_primary_monitor();
+        let monitors = Gdk.Display.get_default().get_monitors();
```

Comment:
GDK 4 has made changes here. We may need to just store the Shell's current idea of the available monitors in a preference and use that here to determine the setting...

Something like...

`shell-available-monitors`
`shell-primary-monitor`

(stored from the extension side of things)

Reply:
I've preferred another approach here, by adding a simple support for `org.gnome.Mutter.DisplayConfig` is enough to make it working as the shell would do.

Reply:
This is now implemented in https://github.com/micheleg/dash-to-dock/pull/1402/commits/714c24a772929eb9e829de4ff72838e3d57f4f27